gridstatus/ieso.py

```diff
@@ -2722,3 +2723,128 @@ def get_transmission_outages_planned(
         ).reset_index(drop=True)
 
         return data
+
+    @support_date_range(frequency="DAY_START")
+    def get_in_service_transmission_limits(
+        self,
+        date: str | pd.Timestamp | tuple[pd.Timestamp, pd.Timestamp],
+        end: pd.Timestamp | None = None,
+        verbose: bool = False,
+    ):
+        if date == "latest":
+            url = f"{PUBLIC_REPORTS_URL_PREFIX}/TxLimitsAllInService0to34Days/PUB_TxLimitsAllInService0to34Days.xml"
+            date = pd.Timestamp.now(tz=self.default_timezone)
+        else:
+            url = f"{PUBLIC_REPORTS_URL_PREFIX}/TxLimitsAllInService0to34Days/PUB_TxLimitsAllInService0to34Days_{date.strftime('%Y%m%d')}.xml"
+
+        xml_content = self._request(url, verbose).text
+
+        return self._process_transmission_limits(
+            xml_content,
+        )
+
+    @support_date_range(frequency="DAY_START")
+    def get_outage_transmission_limits(
+        self,
+        date: str | pd.Timestamp | tuple[pd.Timestamp, pd.Timestamp],
+        end: pd.Timestamp | None = None,
+        verbose: bool = False,
+    ):
+        if date == "latest":
+            urls = [
+                f"{PUBLIC_REPORTS_URL_PREFIX}/TxLimitsOutage0to2Days/PUB_TxLimitsOutage0to2Days.xml",
+                f"{PUBLIC_REPORTS_URL_PREFIX}/TxLimitsOutage3to34Days/PUB_TxLimitsOutage3to34Days.xml",
+            ]
+        else:
+            urls = [
+                f"{PUBLIC_REPORTS_URL_PREFIX}/TxLimitsOutage0to2Days/PUB_TxLimitsOutage0to2Days_{date.strftime('%Y%m%d')}.xml",
+                f"{PUBLIC_REPORTS_URL_PREFIX}/TxLimitsOutage3to34Days/PUB_TxLimitsOutage3to34Days_{date.strftime('%Y%m%d')}.xml",
+            ]
+
+        data_list = []
+        for url in urls:
+            xml_content = self._request(url, verbose).text
+            data_list.append(self._process_transmission_limits(xml_content))
+
+        data = pd.concat(data_list)
+
+        # Drop rows duplicated on every column except Publish Time
+        data = data.drop_duplicates(
+            subset=[c for c in data.columns if c != "Publish Time"],
+            keep="last",
+        ).reset_index(drop=True)
+
+        return data
+
+    def _process_transmission_limits(self, xml_content: str) -> pd.DataFrame:
+        parser = lxml_etree.XMLParser(remove_blank_text=True)
+        tree = lxml_etree.fromstring(xml_content.encode(), parser)
+
+        ns = {"ns": "http://www.ieso.ca/schema"}
+
+        created_at = pd.Timestamp(
+            str(tree.xpath("//ns:CreatedAt/text()", namespaces=ns)[0]),
+        ).tz_localize(
+            self.default_timezone,
+        )
+
+        data = []
+
+        facility_types = {"Internal": "Internal", "Intertie": "Intertie"}
+
+        for xml_type, output_type in facility_types.items():
```

Comment:
do we need a dictionary with identical keys and values? 

```suggestion

        for facility_type in ["Internal", "Intertie"]:
```

and then all the references in the loop can be simply `facility_type`.

Reply:
Whoops, these used to have different keys and values. Updated to use a list.